twingate/client_group.go

```diff
@@ -214,3 +233,60 @@ func (client *Client) deleteGroup(ctx context.Context, groupID graphql.ID) error
 
 	return nil
 }
+
+type GroupsFilter struct {
+	Name     *string
+	Type     *string
+	IsActive *bool
+}
+
+func (f *GroupsFilter) HasName() bool {
+	return f != nil && f.Name != nil && *f.Name != ""
+}
+
+func (f *GroupsFilter) Match(group *Group) bool {
+	if f == nil {
+		return true
+	}
+
+	if f.Type != nil && *f.Type != string(group.Type) {
+		return false
+	}
+
+	if f.IsActive != nil && *f.IsActive != bool(group.IsActive) {
+		return false
+	}
+
+	return true
+}
+
+func (client *Client) filterGroups(ctx context.Context, filter *GroupsFilter) ([]*Group, error) {
+	var (
+		groups []*Group
+		err    error
+	)
+
+	if !filter.HasName() {
+		groups, err = client.readGroups(ctx)
+	} else {
+		groups, err = client.readGroupsByName(ctx, *filter.Name)
+	}
+
+	if err != nil {
+		if errors.Is(err, ErrGraphqlResultIsEmpty) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	var filtered []*Group
+
+	for _, g := range groups {
+		if filter.Match(g) {
```

Comment:
@alexmensch did we agree not to allow filtering by `name` until server api supports it?